eth_tester/utils/filters.py

```diff
@@ -52,16 +52,21 @@ def remove(self, *values):
             values_to_remove = values
 
         queued_values = self.get_changes()
```

Comment:
This should be moved to inside of the `if` block. and the `else` block at the bottom should be dropped.

Reply:
Basically, abort doing anything if `values_to_remove` is empty?

```python
    def remove(self, *values):
        try:
            values_to_remove = set(values)
        except TypeError:
            # log filters are dicts which are not hashable
            values_to_remove = values

        if values_to_remove:
            queued_values = self.get_changes()
            self.values = [
                value
                for value
                in self.get_all()
                if value not in values_to_remove
            ]
            for value in queued_values:
                if value in values_to_remove:
                    continue
                self.queue.put_nowait(value)
```

Reply:
Alternative suggestion:
```python
    def remove(self, *values):
        if not values:
            # Nothing to do!
            return

        try:
            values_to_remove = set(values)
        except TypeError:
            # log filters are dicts which are not hashable
            values_to_remove = values

        queued_values = self.get_changes()
        self.values = [
            value
            for value
            in self.get_all()
            if value not in values_to_remove
        ]
        for value in queued_values:
            if value in values_to_remove:
                continue
            self.queue.put_nowait(value)
```

Reply:
Later suggestion is even better.